util/chplenv/chpl_llvm.py

```diff
@@ -870,15 +870,17 @@ def get_host_link_args():
                 # and -Wl,-syslibroot seems to have no effect
                 system.append("-L" + os.path.join(sdkroot, "usr", "lib"))
 
-        # Decide whether to try to link statically or dynamically.
-        # Future work: consider using 'llvm-config --shared-mode'
-        # to make this choice.
+
+        # For LLVM version 11 and older, there was a problem where
+        # 'llvm-config' did not work properly on Mac OS X, so work around
+        # that by using static linking.
+        llvm_version = get_llvm_version()
         host_platform = chpl_platform.get('host')
-        if host_platform == 'darwin':
+        if host_platform == 'darwin' and int(llvm_version) <= 11:
```

Comment:
I think this is taking a `major.minor` string and casting to an `int`. Is that safe or should we be using something more principled to compare version numbers? I can't think of how this would go wrong.. but I always get nervous doing version comparisons.

Reply:
We would get a Python exception e.g.

```
>>> int("11.3")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '11.3'
```

However, at the moment, these LLVM versions *happen* to be lacking the `.`
I could shift it to `and llvm_version == '11'` if you like.

Reply:
Yeah, I like `and llvm_version == '11'`. If that ever goes wrong, we're only breaking that platform and not this check everywhere.